streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamsProducerTest.java

```diff
@@ -244,47 +247,41 @@ public void shouldCreateProducer() {
     @Test
     public void shouldForwardCallToPartitionsFor() {
         final List<PartitionInfo> expectedPartitionInfo = Collections.emptyList();
-        expect(mockedProducer.partitionsFor("topic")).andReturn(expectedPartitionInfo);
-        replay(mockedProducer);
+        when(mockedProducer.partitionsFor(topic)).thenReturn(expectedPartitionInfo);
 
         final List<PartitionInfo> partitionInfo = streamsProducerWithMock.partitionsFor(topic);
 
         assertThat(partitionInfo, sameInstance(expectedPartitionInfo));
-        verify(mockedProducer);
+        verify(mockedProducer).partitionsFor(topic);
     }
 
     @Test
     public void shouldForwardCallToFlush() {
         mockedProducer.flush();
```

Comment:
```suggestion
        doNothing().when(mockedProducer).flush();
```

Reply:
I feel like here it's more natural to use verify instead of doNothing.